### PR TITLE
add login & register UI without validation

### PR DESCRIPTION
**Pull request by HAYATO-Kz**
- Add login & register UI (without validation state)
- Add button component
- Add input component (without error state)
- Prepare the main screen for auth
- Add register form validation

### CODE REVIEW

File: components/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      onPress={onPress}
+      outline={outline}
+      block={block}
+      disabled={disabled}>
+      <ButtonText outline={outline}>{text}</ButtonText>
+    </ButtonWithStyled>
+  );
+};
+
+const ButtonWithStyled = styled.TouchableOpacity<{
```
Comment (NokKbl) [inline components/Button.tsx:40]: Can we separate style and code file later on?

File: ios/demo/Info.plist [hunk 2]
```diff
@@ -43,6 +39,10 @@
 	</dict>
 	<key>NSLocationWhenInUseUsageDescription</key>
 	<string></string>
+	<key>UIAppFonts</key>
+	<array>
+		<string>Entypo.ttf</string>
```
Comment (NokKbl) [inline ios/demo/Info.plist:44]: Why don't you apply the same fonts that used in the design?

File: screens/Home.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+const Home = (props: HomeProps) => {
+  const {navigation} = props;
+
+  return (
+    <HomeContainer>
+      <Logo source={require('../images/logo.png')} />
+      <Mascot source={require('../images/mascot.png')} />
+      <ButtonContainer>
+        <Button
+          text="USE WITHOUT LOGIN"
```
Comment (NokKbl) [inline screens/Home.tsx]:
```suggestion
          text="USE WITHOUT LOG IN"
```

File: screens/Home.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+          }}
+          block
+        />
+        <DividerRow>
+          <Divider>OR</Divider>
+        </DividerRow>
+        <SecondaryButtonContainer>
+          <SecondartButtonWrapper>
+            <Button
+              text="LOGIN"
```
Comment (NokKbl) [inline screens/Home.tsx]:
```suggestion
              text="LOG IN"
```

File: screens/Login.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    <Container>
+      <SafeAreaView />
+      <Icon
+        type="Entypo"
+        name="chevron-left"
+        style={{color: 'black'}}
+        onPress={() => {
+          navigation.navigate('Home');
+        }}></Icon>
+      <HeaderText>Login</HeaderText>
```
Comment (NokKbl) [inline screens/Login.tsx]:
```suggestion
      <HeaderText>Log in</HeaderText>
```

File: screens/Register.tsx [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+          <View>
+            <Input label="Username" onChangeText={handleChange('username')} />
+            <Input label="Email" onChangeText={handleChange('email')} />
+            <Input
+              label="Password"
+              onChangeText={handleChange('password')}
+              secureTextEntry={true}
+            />
+            <ButtonWrapper>
+              <Button text="SIGN UP" block onPress={handleSubmit} />
```
Comment (NokKbl) [inline screens/Register.tsx]:
```suggestion
              <Button text="REGISTER NOW" block onPress={handleSubmit} />
```

File: components/Button.tsx (hunk 1 above)
Comment (HAYATO-Kz) [reply components/Button.tsx:40]: Yes. After the refactor phase structure will be like

       root
       |- components
       |      |- index.ts
       |      |- Button
       |           |- Button.tsx
       |           |- Button.style.ts
       |- screens
              |- index.ts
              |- Home
                   |- Home.tsx
                   |- Home.style.ts

File: screens/Review.tsx [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+   * @param lat latitude's blog
+   * @param long longitude's blog
+   */
+  const openInGoogleMap = async (lat: number, long: number) => {
+    const url = `https://www.google.com/maps/dir/?api=1&destination=${lat},${long}&dir_action=navigate`;
+    const supported = await Linking.canOpenURL(url);
+    if (supported) {
+      await Linking.openURL(url);
+    } else {
+      console.log(`Don't know how to open this URL: ${url}`);
```
Comment (NokKbl) [inline screens/Review.tsx]:
```suggestion
      console.log(`Invalid URL: ${url}`);
```

File: screens/Review.tsx [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+          </Button>
+        </Right>
+      </Header>
+      <Content style={{padding: 20}}>
+        <Grid>
+          <Row style={{marginBottom: 20}}>
+            <Title>{source.title}</Title>
+          </Row>
+          <Row style={{marginVertical: 5}}>
+            <Text>Reviewer : {source.nickname}</Text>
```
Comment (NokKbl) [inline screens/Review.tsx]:
```suggestion
            <Text>REVIEWER : {source.nickname}</Text>
```

File: ios/demo/Info.plist (hunk 2 above)
Comment (HAYATO-Kz) [reply ios/demo/Info.plist:44]: I looking to add the theme to this project after finishing add the authentication flow. so I'm not sure that I need to set the font in this file or in the theme file

File: components/Button.tsx (hunk 1 above)
Comment (NokKbl) [reply components/Button.tsx:40]: Gotcha

File: ios/demo/Info.plist (hunk 2 above)
Comment (NokKbl) [reply ios/demo/Info.plist:44]: @HAYATO-Kz Please let me know when you sure.

File: screens/Home.tsx (hunk 3 above)
Comment (HAYATO-Kz) [reply screens/Home.tsx]: DONE

File: screens/Home.tsx (hunk 4 above)
Comment (HAYATO-Kz) [reply screens/Home.tsx]: DONE

File: screens/Login.tsx (hunk 5 above)
Comment (HAYATO-Kz) [reply screens/Login.tsx]: DONE

File: screens/Register.tsx (hunk 6 above)
Comment (HAYATO-Kz) [reply screens/Register.tsx]: DONE

File: screens/Review.tsx (hunk 7 above)
Comment (HAYATO-Kz) [reply screens/Review.tsx]: DONE

File: screens/Review.tsx (hunk 8 above)
Comment (HAYATO-Kz) [reply screens/Review.tsx]: DONE

File: screens/Login.tsx [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+          <View>
+            <Input label="email" onChangeText={handleChange('email')} />
+            <Input
+              label="password"
+              onChangeText={handleChange('password')}
+              secureTextEntry={true}
+            />
+            <ButtonWrapper>
+              <Button
+                text="LOGIN"
```
Comment (NokKbl) [inline screens/Login.tsx]:
```suggestion
                text="LOG IN"
```

**Review by NokKbl — APPROVED** [review]
LGTM for now

**Pull request merged**